### ISSUE TITLE
Outh2proxy with keycloak as a backend jwt token verifier 

### ISSUE THREAD

**Issue by miritsadon** [oauth2-proxy/oauth2-proxy#1274]
## Expected Behavior
 
I should be able to configure the service to return a json body format instead of html page.
 

## Current Behavior

Receiving html page with 401 response

## Steps to Reproduce (for bugs)

1.  
2.  
3.  
4.  

## Your Environment
 K8 cluster with nginx ingress controller 
Keycloak, oauth2proxy  service and an additional backed service using nginx annotations: auth-url to redirect the authentication to oauth2proxy service. 

- Version used:

**Comment by JoelSpeed** [comment]
I think if the request to OAuth2 Proxy has a header suggesting it wants a json response, then OAuth2 Proxy does handle this correctly.

Have you tried making a request with the `Accept: applciation/json` header?

**Comment by miritsadon** [comment]
Yes all my original calls are with the header. But i am getting an html
response. I wander if the redirect request to oauth is missing the header.
But not sure how to configure that.

**Comment by JoelSpeed** [comment]
Looking at your setup, I think you need to handle this in Nginx. When you set up the `auth-url`, that will always reply with a 202 or 401, Nginx reacts to the error type by then sending you to the `auth-signin` URL. What did you set for this value? If you set it to OAuth2 Proxies login page, then that explains why you're getting a HTML page rendered.

You need Nginx to react to the 401 and return a 401 page instead for this kind of request.
